Scripts/run-tests.sh

```diff
@@ -0,0 +1,57 @@
+#!/bin/bash
+
+set -o pipefail
+exitCode=0
+
+function checkStatus {
+    if [[ $exitCode == 0 && $1 != 0 ]]; then
+        exitCode=$1
+    fi
+}
+
+function uploadCodecov {
+    bash <(curl -s https://codecov.io/bash)
+}
+
```

Comment:
My recommendation is to set default values in the script as in (1), but realize that this will only set the values if they are not already set in the environment.  So we can (and should) keep the values in the `travis.yml` without concern for whether they match the default values in the script.  The point of the default values is simply to avoid unpleasant errors if you run the script locally but forgot to set the variables in your environment.